File: 03_Arrays/Easy/C++/10_consecutive_ones.cpp
#include <bits/stdc++.h>
using namespace std;

int consecutiveOnes(vector<int>& arr){
    int maxi = 0;
    int n = arr.size();
    int cnt = 0;
    for(int i=0; i<n; ++i){
        if(arr[i] == 1){
            ++cnt;
        }
        else {
            cnt = 0;
        }
        maxi = max(maxi, cnt);
    }

    return maxi;
}

int main(int argc, char* argv[]){
  vector<int> v;

  int size;
  cin >> size;

  // enter sorted array
  for(int i=0; i<size; ++i){
    int a;
    cin >> a;
    v.push_back(a);
  }

  int maxOnes = consecutiveOnes(v);

  cout << "Maximium no. of consecutive ones: " << maxOnes;
return 0;
}

File: 03_Arrays/Easy/Python/10_consecutive_ones.py
from typing import List

class Solution:
  def consecutive_ones(self, arr: List[int]) -> int:
    cnt = 0
    maxi = 0
    n = len(arr)

    for i in range(n):
      if arr[i] == 1:
        cnt += 1
      else:
        cnt = 0
      maxi = max(maxi, cnt)
    return maxi
  
size = int(input("size: "))
arr = list(map(int, input("Enter sorted array: ").strip().split()[:size]))

sol = Solution()

print(sol.consecutive_ones(arr))